### PR TITLE
Feature Request: add `[advancement]` to `[modify_unit_type]`

### PR DESCRIPTION
**Pull request by knyghtmare**
Addresses #9309

### CODE REVIEW

**Comment by gfgtdf** [comment]
This does nothing except adding it to the schema, it doesn't add any functionality at all. (This being marked as ready-for-testing I'm assuming the pr is finished)

**Comment by knyghtmare** [comment]
> This does nothing except adding it to the schema, it doesn't add any functionality at all. (This being marked as ready-for-testing I'm assuming the pr is finished)

Which .cpp files have to modified to get this working?
I checked the files but dont see how it's not working.

```cpp
for(const config& modify_unit_type : level_.child_range("modify_unit_type")) {
        unit_types.apply_scenario_fix(modify_unit_type);
    }
```

```cpp
for(const config& modlua : cfg->child_range("modify_unit_type")) {
            starting_point_.add_child_at_total("modify_unit_type", modlua, pos++);
        }
```

Cannot find any other instances of the tag here.

**Comment by knyghtmare** [comment]
- [play_controller.cpp](https://github.com/wesnoth/wesnoth/blob/711b77b7e25c2d94c0dc1611801210b5179b696b/src/play_controller.cpp#L167)
- [saved_game.cpp](https://github.com/wesnoth/wesnoth/blob/711b77b7e25c2d94c0dc1611801210b5179b696b/src/saved_game.cpp#L366)

**Comment by knyghtmare** [comment]
> You should create unit test for this.

How to "create unit test"?

**Comment by ProditorMagnus** [comment]
Add scenario that tests it to wml tests.

File: src/units/types.cpp [hunk 1]
```diff
@@ -1394,6 +1394,26 @@ void unit_type::apply_scenario_fix(const config& cfg)
 			boost::remove_erase(advances_to_, str);
 		}
 	}
+	// support advancement AMLA tags here
+	// TODO: I think I badly screwed up the if condition here
+	// disclaimer: I dont how to code cpp, I know Python and SQL.
+	if (auto attr = cfg.child_range("advancement")) {
```
Comment (CelticMinstrel) [inline src/units/types.cpp:1400]: You want has_child, not child_range.

Also, I don't see any code here that actually adds the advancement. You're adding to the "abilities that can be obtained by advancements" list, but the advancement itself hasn't been added.

Comment (knyghtmare) [reply src/units/types.cpp:1400]: > You're adding to the "abilities that can be obtained by advancements" list, but the advancement itself hasn't been added.

What does an adding advancement line look like? (I don't know Cpp)

Comment (newfrenchy83) [reply src/units/types.cpp:1400]: Look at this https://github.com/wesnoth/wesnoth/pull/9353

**Comment by knyghtmare** [comment]
Closed in favor of https://github.com/wesnoth/wesnoth/pull/9353

**Pull request closed without merging**